src/dmd/cppmangle.d

```diff
@@ -138,6 +151,40 @@ private final class CppMangleVisitor : Visitor
         return -1;
     }
 
+    /**
+     * Overload which accepts a Namespace
+     *
+     * It is very common for large C++ projects to have multiple files sharing
+     * the same `namespace`. If any D project adopts the same approach
+     * (e.g. separating data structures from functions), it will lead to two
+     * `Nspace` object being instantiated, with different address.
+     * At the same time, we cannot compare just any Dsymbol via identifier,
+     * because it messes with templates.
+     *
+     * See_Also:
+     *  https://issues.dlang.org/show_bug.cgi?id=18922
+     *
+     * Params:
+     *   ns = C++ namespace to do substitution for
+     *
+     * Returns:
+     *  Index of the entry, if found, or `-1` otherwise
+     */
+    int find(Nspace ns)
+    {
+        foreach (i, component; components)
+        {
+            if (ns == component)
+                return cast(int)i;
```

Comment:
Is this cast really necessary? I would argue it is not, but maybe I am wrong

Reply:
Well it was there before, I just followed the existing style.
I would much prefer to return `components.length` and use `size_t`, but I'll leave that for another refactor...